torchvision/datasets/widerface.py

```diff
@@ -0,0 +1,183 @@
+from .vision import VisionDataset
+
+
+class WIDERFace(VisionDataset):
+    """`WIDERFace <http://shuoyang1213.me/WIDERFACE/>`_ Dataset.
+
+    Args:
+        root (string): Root directory where images and annotations are downloaded to.
+            Expects the following folder structure if download=False:
+                <root>
+                    └── widerface
+                        ├── wider_face_split ('wider_face_split.zip' if compressed)
+                        ├── WIDER_train ('WIDER_train.zip' if compressed)
+                        ├── WIDER_val ('WIDER_val.zip' if compressed)
+                        └── WIDER_test ('WIDER_test.zip' if compressed)
+        split (string): The dataset split to use. One of {``train``, ``val``, ``test``}.
+            Defaults to ``train``.
+        transform (callable, optional): A function/transform that  takes in a PIL image
+            and returns a transformed version. E.g, ``transforms.RandomCrop``
+        target_transform (callable, optional): A function/transform that takes in the
+            target and transforms it.
+        download (bool, optional): If true, downloads the dataset from the internet and
+            puts it in root directory. If dataset is already downloaded, it is not
+            downloaded again.
+    """
+
+    BASE_FOLDER = "widerface"
+    FILE_LIST = [
+        # File ID                        MD5 Hash                            Filename
+        ("0B6eKvaijfFUDQUUwd21EckhUbWs", "3fedf70df600953d25982bcd13d91ba2", "WIDER_train.zip"),
+        ("0B6eKvaijfFUDd3dIRmpvSk8tLUk", "dfa7d7e790efa35df3788964cf0bbaea", "WIDER_val.zip"),
+        ("0B6eKvaijfFUDbW4tdGpaYjgzZkU", "e5d8f4248ed24c334bbd12f49c29dd40", "WIDER_test.zip")
+    ]
+    ANNOTATIONS_FILE = ("http://mmlab.ie.cuhk.edu.hk/projects/WIDERFace/support/bbx_annotation/wider_face_split.zip",
+                        "0e3767bcf0e326556d407bf5bff5d27c",
+                        "wider_face_split.zip")
+
+    def __init__(
+            self,
+            root: str,
+            split: str = "train",
+            transform: Optional[Callable] = None,
+            target_transform: Optional[Callable] = None,
+            download: bool = False,
+    ) -> None:
+        super(WIDERFace, self).__init__(root=os.path.join(root, self.BASE_FOLDER),
+                                        transform=transform,
+                                        target_transform=target_transform)
+        # check arguments
+        self.split = verify_str_arg(split, "split", ("train", "val", "test"))
+
+        if download:
+            self.download()
+
+        if not self._check_integrity():
+            raise RuntimeError("Dataset not found or corrupted. " +
+                               "You can use download=True to download and prepare it")
+
+        # process dataset
+        # dataset will be stored as a list of dict objects (img_info)
+        self.img_info: Any = []
```

Comment:
I've removed those two comment lines entirely as the function names are pretty self-documenting.

Reply:
My bad for not being precise enough: I meant the type annotation `Any`. At least it should be `List[Any]` but could probably be even more precise.

Reply:
Would this be precise enough?

```suggestion
        self.img_info: List[Dict[str, Any]] = []
```

Reply:
Depends :wink: What type do the values of the dicts have? As a rule of thumb one should only use `Any` if it is indeed not important or simply not expressible.

In this case here IMO this would be the correct annotation:

```suggestion
        self.img_info: List[Dict[str, Union[str, Dict[str, torch.Tensor]]]] = []
```

Reply:
You are correct. I've made the change you suggested. Resolving for conversation for now but feel free to open back up if you have any other feedback.